mocha_test/mapValues.js

```diff
@@ -0,0 +1,92 @@
+var async = require('../lib');
+var expect = require('chai').expect;
+var assert = require('assert');
+
+describe('mapValues', function () {
+    var obj = {a: 1, b: 2, c: 3};
+
+    context('mapValuesLimit', function () {
+        it('basics', function (done) {
+            var running = 0;
+            var concurrency = {
+                a: 2,
+                b: 2,
+                c: 1
+            };
+            async.mapValuesLimit(obj, 2, function (val, key, next) {
+                running++;
+                async.setImmediate(function () {
+                    expect(running).to.equal(concurrency[key]);
```

Comment:
We'd have to refactor other tests to be like this style, tracking concurrency and verifying things finish in a certain order, but it might be worth it :sweat: